### PR TITLE
Add Singletons page

### PR DESCRIPTION
**Pull request by dalemyers**

### CODE REVIEW

File: Singletons [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+```swift
+// Good
+class MyManager {
+  static let shared = MyManager()
+  var someVar: String = "Hello"
+}
+
+// Bad
+class MyManager {
+  static var someVar: String = "Hello"
```
Comment (cy255) [inline Singletons]: Could we have a different good / bad example and merge the good / bad example with the one below? 
I think there's value in having a static property and it is useful in certain cases (e.g. class identifier) and in that case a singleton is not needed.

**Review by cy255 — CHANGES_REQUESTED** [review]

File: Singletons (hunk 1 above)
Comment (dalemyers) [reply Singletons]: I'm not really following? I agree that static properties are useful (and indeed required) in certain cases. I think this is clear though that it only applies to singletons, and not to static properties in general?

File: Singletons.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  var someVar: String = "Hello"
+}
+
+// Bad
+class MyManager {
+  static var someVar: String = "Hello"
+}
+```
+
+This is enforced for 2 main reasons:
```
Comment (cy255) [inline Singletons.md]: This section is essentially the "Rationale" section. Could we follow the same format as other conventions?

## Rationale
This makes it clear that this class is a singleton and should be used that way, and it allows easier testing as new versions can be instantiated.

File: Singletons.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+1. It makes it clear that this class is a singleton and should be used that way.
+2. It allows easier testing (as new versions can be instantiated)
+
+
+Here's an example:
```
Comment (cy255) [inline Singletons.md]: My original comment was outdated, continuing the discussions here.
Could we have an examples section and use the following example as the "good" example and remove the examples from line 5 to line 16? 

##Examples

```swift
// bad: the singleton has static properties
public class OurDefaults {
    // Called throughout the application
    public static let shared = OurDefaults(userDefaults: .standard)

    // Called during test case set up
    // Initializer also makes it clear how many dependencies this class has.
    init(userDefaults: UserDefaults) {
        self.userDefaults = userDefaults
    }

    private static let userDefaults: UserDefaults
 }

// good: the singleton has instance properties
public class OurDefaults {
    // Called throughout the application
    public static let shared = OurDefaults(userDefaults: .standard)

    // Called during test case set up
    // Initializer also makes it clear how many dependencies this class has.
    init(userDefaults: UserDefaults) {
        self.userDefaults = userDefaults
    }

    private let userDefaults: UserDefaults
 }
 ```

**Review by cy255 — CHANGES_REQUESTED** [review]

File: Singletons.md (hunk 3 above)
Comment (dalemyers) [reply Singletons.md]: I've got something slightly different in the latest version, but I think it covers the same points.

**Review by cy255 — APPROVED** [review]

**Pull request merged**